python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -206,6 +225,88 @@ def run(self, exposure, sources, background):
+            Struct with various psf stats.
+        """
+        psfStats = pipeBase.Struct(nPsfStar=0,
+                                   psfStarDeltaE1Median=float(np.nan),
+                                   psfStarDeltaE2Median=float(np.nan),
+                                   psfStarDeltaE1Scatter=float(np.nan),
+                                   psfStarDeltaE2Scatter=float(np.nan),
+                                   psfStarDeltaSizeMedian=float(np.nan),
+                                   psfStarDeltaSizeScatter=float(np.nan),
+                                   psfStarScaledDeltaSizeScatter=float(np.nan))
+
+        if sources is None:
+            # No sources are available (as in some tests)
+            return psfStats
+
+        names = sources.schema.getNames()
+        if self.config.starSelection not in names or self.config.starShape + '_flag' not in names:
+            # The source catalog does not have the necessary fields (as in some tests)
+            return psfStats
+
+        mask = sources[self.config.starSelection] & (~sources[self.config.starShape + '_flag'])
+        nPsfStar = mask.sum()
+
+        if nPsfStar == 0:
+            # This is all bad.
+            return psfStats
+
+        starXX = sources[self.config.starShape + '_xx'][mask]
+        starYY = sources[self.config.starShape + '_yy'][mask]
+        starXY = sources[self.config.starShape + '_xy'][mask]
+        psfXX = sources[self.config.psfShape + '_xx'][mask]
+        psfYY = sources[self.config.psfShape + '_yy'][mask]
+        psfXY = sources[self.config.psfShape + '_xy'][mask]
+
+        starSize = (starXX*starYY - starXY**2.)**0.25
+        starE1 = (starXX - starYY)/(starXX + starYY)
+        starE2 = 2*starXY/(starXX + starYY)
+        starSizeMedian = np.median(starSize)
+
+        psfSize = (psfXX*psfYY - psfXY**2)**0.25
+        psfE1 = (psfXX - psfYY)/(psfXX + psfYY)
+        psfE2 = 2*psfXY/(psfXX + psfYY)
+
+        psfStarDeltaE1Median = np.median(starE1 - psfE1)
+        psfStarDeltaE1Scatter = sigmaMad(starE1 - psfE1, scale='normal')
+        psfStarDeltaE2Median = np.median(starE2 - psfE2)
+        psfStarDeltaE2Scatter = sigmaMad(starE2 - psfE2, scale='normal')
+
+        psfStarDeltaSizeMedian = np.median(starSize - psfSize)
+        psfStarDeltaSizeScatter = sigmaMad(starSize - psfSize, scale='normal')
+        psfStarScaledDeltaSizeScatter = psfStarDeltaSizeScatter/starSizeMedian**2.
+
+        psfStats.nPsfStar = int(nPsfStar)
+        psfStats.psfStarDeltaE1Median = float(psfStarDeltaE1Median)
+        psfStats.psfStarDeltaE2Median = float(psfStarDeltaE2Median)
+        psfStats.psfStarDeltaE1Scatter = float(psfStarDeltaE1Scatter)
+        psfStats.psfStarDeltaE2Scatter = float(psfStarDeltaE2Scatter)
+        psfStats.psfStarDeltaSizeMedian = float(psfStarDeltaSizeMedian)
+        psfStats.psfStarDeltaSizeScatter = float(psfStarDeltaSizeScatter)
+        psfStats.psfStarScaledDeltaSizeScatter = float(psfStarScaledDeltaSizeScatter)
```

Comment:
Just out of curiosity, why is all this casting needed?

Reply:
I've added a comment that we must cast to python quantities rather than numpy so that these can get serialized as yaml.